diff --git a/src/xrt/compositor/multi/comp_multi_system.c b/src/xrt/compositor/multi/comp_multi_system.c
index 5a4f35b..cdae379 100644
--- a/src/xrt/compositor/multi/comp_multi_system.c
+++ b/src/xrt/compositor/multi/comp_multi_system.c
@@ -657,7 +657,6 @@ system_compositor_set_z_order(struct xrt_system_compositor *xsc, struct xrt_comp
 
 	//! @todo Locking?
 	mc->state.z_order = z_order;
-	os_mutex_unlock(&msc->list_and_timing_lock);
 
 	return XRT_SUCCESS;
 }
